src/simsopt/geo/boozersurface.py

```diff
@@ -40,17 +40,48 @@ class BoozerSurface(Optimizable):
 
     where Newton is used to solve the first order necessary conditions for optimality.
     """
-
-    def __init__(self, biotsavart, surface, label, targetlabel):
+    
+    def __init__(self, biotsavart, surface, label, targetlabel, constraint_weight=None):
         super().__init__(depends_on=[biotsavart])
         self.biotsavart = biotsavart
         self.surface = surface
         self.label = label
         self.targetlabel = targetlabel
+        self.constraint_weight = constraint_weight
         self.need_to_run_code = True
 
     def recompute_bell(self, parent=None):
         self.need_to_run_code = True
+    
+    def run_code(self, boozer_type, iota, G=None, verbose=True):
```

Comment:
I suggest making tol and maxiter keyword arguments (with defaults) so the user can adjust if necessary.

Reply:
I've added support for solver options now, thanks!